draft-dss-star.md

```diff
@@ -101,200 +121,640 @@ In this document we describe one such system, namely Distributed Secret Sharing
+  representation of input `element`, denoted `blinded_element`, along with a value to revert
+  the blinding process, denoted `blind`.
+- Finalize(element, blind, evaluated_element): Finalize the OPRF evaluation using input `element`,
+  random inverter `blind`, and evaluation output `evaluated_element`, yielding output `oprf_output`.
+
+Moreover, the following OPRF server APIs are used:
+
+- Evaluate(k, blinded_element): Evaluate blinded input element `blinded_element` using
+  input key `k`, yielding output element `evaluated_element`. This is equivalent to
+  the Evaluate function described in {{OPRF, Section 3.3.1}}, where `k` is the private key parameter.
+- DeriveKeyPair(seed, info): Derive a private and public key pair deterministically
+  from a seed and info parameter, as described in {{OPRF, Section 3.2}}.
+
+Finally, this specification makes use of the following shared APIs and parameters:
+
+- SerializeElement(element): Map input `element` to a fixed-length byte array `buf`.
+- DeserializeElement(buf): Attempt to map input byte array `buf` to an OPRF group element.
+  This function can raise a DeserializeError upon failure; see {{OPRF, Section 2.1}}
+  for more details.
+- Noe: The size of a serialized OPRF group element output from SerializeElement.
+
+This specification uses the base (non-verifiable) OPRF from {{OPRF, Section 3}} with the
+OPRF(ristretto255, SHA-512) as defined in {{OPRF, Section 4.1.1}}.
+
+## Key Derivation Function {#deps-kdf}
+
+A Key Derivation Function (KDF) is a function that takes some source of initial
+keying material and uses it to derive one or more cryptographically strong keys.
+This specification uses a KDF with the following API and parameters:
+
+- Extract(salt, ikm): Extract a pseudorandom key of fixed length `Nx` bytes from
+  input keying material `ikm` and an optional byte string `salt`.
+- Expand(prk, info, L): Expand a pseudorandom key `prk` using the optional string `info`
+  into `L` bytes of output keying material.
+- Nx: The output size of the `Extract()` function in bytes.
+
+This specification uses HKDF-SHA256 {{!HKDF=RFC5869}} as the KDF function, where Nx = 32.
 
-## Objective
+## Key-Committing Authenticated Encryption with Associated Data {#deps-aead}
 
-In STAR, clients generate `encrypted` measurements, that they send to a single untrusted Aggregation Server. In a given amount of time, if the Aggregation Server receives the same encrypted value from K clients (i.e. K values), the server is able to decrypt the value. This ensures that clients only have their measurements revealed if they are part of a larger crowd. This allows the client to maintain K-anonymity, when paired with mechanisms for removing client-identifying information from their requests.
+A Key-Committing Authenticated Encryption with Associated Data (KCAEAD) scheme is an algorithm
+for encrypting and authenticating plaintext with some additional auxiliary data.
```

Comment:
This seems like it is going to get confusing, because auxiliary data support is explicitly provided in STAR, but the auxiliary data here is referring to additional data that can be included separately with the AEAD encryption process. It looks like you are suggesting that the auxiliary data is appended using the AEAD additional data functionality, rather than within the encrypted ciphertext as it is in the STAR paper, is that correct?

Reply:
No, thats not the intent. This is just defining the syntax of an AEAD, which explicitly includes additional data.